src/ng/directive/select.js

```diff
@@ -70,7 +70,9 @@ var SelectController =
   };
 
   self.unselectEmptyOption = function() {
-    if (self.emptyOption) {
+    // Empty options may be comment nodes if they have been removed by
+    // transclude directives such as `ngIf`
+    if (self.emptyOption && self.emptyOption[0].nodeType === NODE_TYPE_ELEMENT) {
```

Comment:
Isn't there always going to be a comment there, even when `ngIf` is true?
Shouldn't we look for the first non-comment element in `self.emptyOption`?